repondeur/zam_repondeur/models/lecture.py

```diff
@@ -236,3 +240,41 @@ def url_key(self) -> str:
         else:
             partie = ""
         return f"{self.chambre}.{self.session}.{self.num_texte}{partie}.{self.organe}"
+
+    def find_article(self, subdiv: SubDiv) -> Optional[Article]:
+        article: Article
+        for article in self.articles:
+            if article.matches(subdiv):
+                return article
+        return None
+
+    def find_or_create_article(self, subdiv: SubDiv) -> Tuple[Article, bool]:
+        article = self.find_article(subdiv)
+        created = False
+        if article is None:
+            article = Article.create(
+                lecture=self,
+                type=subdiv.type_,
+                num=subdiv.num,
+                mult=subdiv.mult,
+                pos=subdiv.pos,
+            )
+            created = True
+        return article, created
```

Comment:
At that point, does `self.articles` contain the newly created article or does it need a DBSession flush/commit?

Reply:
Yes it does.

That's the beauty of relationships when working with objects (versus just by setting foreign keys).

If I do:
```python
article = Article(lecture=lecture, ...)
```
then this is true:
```python
assert article in lecture.articles
```